pkg/ip/link.go

```diff
@@ -40,9 +41,15 @@ func makeVethPair(name, peer string, mtu int) (netlink.Link, error) {
 	return veth, nil
 }
 
-func makeVeth(name string, mtu int) (peerName string, veth netlink.Link, err error) {
+func makeVeth(namePrefix string, mtu int) (peerName string, veth netlink.Link, err error) {
 	for i := 0; i < 10; i++ {
-		peerName, err = RandomVethName()
+		peerName, err = RandomIfaceName(namePrefix)
```

Comment:
I see, have we agreed/disagreed on the counter-based indexing for interfaces? Could we just use the interfaces kernel ifindex? They should always be unique per netns.